### PR TITLE
New areas script update

### PR DESCRIPTION
**Pull request by mfbenitezp**
1. I did not know how to add new dependencies to the cargo file; thus, I added [cargo-edit ](https://lib.rs/crates/cargo-edit) to add click dependency; I hope that's ok. 
2. I modified the `select_msoas.py `script to make it slightly easier to use.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ bincode = "1.3.3"
 bytes = "1.1.0"
 cap = "0.1.0"
 clap = { version = "3.0.0", features = ["derive"] }
+click = "0.5.4"
```
Comment (dabreegster) [inline Cargo.toml:14]: `Cargo.toml` is only for Rust dependencies. https://crates.io/crates/click isn't what you expected here. :)

The Python scripts in this repo don't use any environment manager like conda or poetry. We could just ask people to run the script inside an environment with click, or we could prescribe conda or poetry, or we could tell people to `pip install click`. Unless we wind up with lots of python scripts in this repo (unlikely), I'm not sure what to do. What's typical in a project you've encountered that has lots of random scripts, some of which have a few dependencies?

File: scripts/select_msoas.py [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
-print('"MSOA11CD"')
-# You must run the SPC pipeline to download this file
-with open('data/raw_data/referencedata/lookUp.csv') as f:
-    for row in csv.DictReader(f):
-        # Modify these to select MSOAs based on different geographies
-        if row['LAD20NM'] == 'Liverpool':
-            print('"{}"'.format(row['MSOA11CD']))
+        with open(newFileNameLocation, 'w') as csvfile:
+            fieldnames = ["MSOA11CD"]
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames, quoting=csv.QUOTE_NONNUMERIC)
```
Comment (dabreegster) [inline scripts/select_msoas.py:28]: This is a very complicated way to print a CSV file with one column. Why not keep the old approach and just do `csvfile.write('"{}"'.format(row['MSOA11CD']))` or similar?

File: scripts/select_msoas.py [hunk 3]
```diff
@@ -1,12 +1,37 @@
 # This script helps you define new study areas. You can write the output to a
-# new file, config/some_region.csv .
-
+# new file, config/some_region.csv.
```
Comment (dabreegster) [inline scripts/select_msoas.py:2]: The second sentence is incorrect now, the script writes output for you

File: scripts/select_msoas.py [hunk 4]
```diff
@@ -1,12 +1,37 @@
...
+@click.option('--lda',
+             default='e.g Liverpool',
+             prompt = "Name of Local Authority District",
+             help="Enter the name of the Local Authority District of your preference, to get the MSOA codes for that area")
+
+def area(lda):
+    newarea= []
+    # You must run the SPC pipeline to download this file
+    click.echo("Querying MSOA codes for, %s!" % lda)
+    newFileNameLocation = "config/"+lda+".csv"
```
Comment (dabreegster) [inline scripts/select_msoas.py:16]: The output filenames don't match existing stuff in `config`. We should pick a consistent style. Do you prefer `west_yorkshire`, `WestYorkshire`, `West_Yorkshire`, something else?

**Review by dabreegster — COMMENTED** [review]
Could you please add `.DS_Store` to `.gitignore` and remove it from this PR? (And also revert the changes to `Cargo.lock` and toml)

File: Cargo.toml (hunk 1 above)
Comment (mfbenitezp) [reply Cargo.toml:14]: - [X] add .DS_Store to .gitignore, I guess I should remove this PR and create a new one for your review?
- [ ]  I dont really know, I dont see the issue of having click here. My constrain here is I dont think people should copy and paste the shell outcome to create a new csv, the tool should make that easier for them. We should provide an easy way to get the new study areas, so I was attempting that.

File: scripts/select_msoas.py (hunk 2 above)
Comment (mfbenitezp) [reply scripts/select_msoas.py:28]: Well this is the way I know, if you think is way complicated, then lets just make it as you've suggested. The important part for me is to write it in a CSV file directly.

File: scripts/select_msoas.py (hunk 3 above)
Comment (mfbenitezp) [reply scripts/select_msoas.py:2]: Ok.

File: scripts/select_msoas.py (hunk 4 above)
Comment (mfbenitezp) [reply scripts/select_msoas.py:16]: We had a similar debate in ASPICS, and I think you suggested to agree on lowcase_newareas e.g west_yorkshire. Then I will do it like this.

**Comment by mfbenitezp** [comment]
I have followed your suggestions and done what you've requested. Please take a look at the added commits.

**Comment by dabreegster** [comment]
@mfbenitezp, please take a look at the changes:

- I reverted the Cargo.lock changes and deleted DS_Store from git
- No more csv.DictWriter at all; it's lots of pointless extra steps to write literally one column
- Transforming the LAD name to a Unix-friendly output filename without spaces. We can further name this whatever we want; two_counties and west_yorkshire_large are just constructs that were useful
- Warn when the input isn't a known LAD at all

The lookup table has a bunch of other columns; we could make this even nicer by asking the user which category they want to search by too. If we wanted, it could even prompt users to select from a list of all valid options for both questions. But I suspect that's overkill. Since you are the one defining new study areas right now, let's make the script convenient for your use case, and not try to guess what hypothetical future users might want to do.

**Comment by dabreegster** [comment]
We're having merge conflicts pulling down locally with git, so I squashed. Merging.

**Pull request merged**